File: src/main/ui/src/models/view/PDUIElementMeta.ts
import type {PDViewMeta} from "@/models/view/PDViewMeta";

export interface PDUIElementMeta extends PDViewMeta {
    labelVisible: boolean;
    colLG: number;
    colMD?: number;
    colSM?: number;
    colOffset?: number;
    unwrapColumn?: boolean;
}

File: src/main/java/de/fhswf/genericapplication/models/view/PDUIElementMeta.java
package de.fhswf.genericapplication.models.view;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.xml.bind.annotation.XmlAttribute;
import javax.xml.bind.annotation.XmlRootElement;

@XmlRootElement(namespace = "view")
public abstract class PDUIElementMeta extends PDViewMeta {
    @XmlAttribute
    @JsonProperty
    private boolean labelVisible = true;

    @XmlAttribute
    @JsonProperty
    private int colSM = 12;

    @XmlAttribute
    @JsonProperty
    private int colMD = 12;

    @XmlAttribute
    @JsonProperty
    private int colLG = 12;

    @XmlAttribute
    @JsonProperty
    private int colOffset = 0;
}
